FinSearchUnified/Tests/Bundle/ProductNumberSearchTest.php

```diff
@@ -106,31 +136,221 @@ public function testProductNumberSearchImplementation($isFetchCount, $isUseShopS
+                'xmlResponse' => $xmlResponse,
+                'expectedResults' => [
+                    TreeFacetResult::class
+                ]
+            ]
         ];
     }
+
+    public function missingFilterProvider()
+    {
+        $xmlResponse = $this->getXmlResponse();
+        $filters = $xmlResponse->addChild('filters');
+
+        $this->setCategoryFilter($filters);
+        $this->setVendorFilter($filters, 'select', false);
+
+        return [
+            'Filter is missing in response' => [
+                'xmlResponse' => $xmlResponse,
+                'expectedResults' => [
+                    TreeFacetResult::class
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider allFiltersProvider
+     * @dataProvider facetWithNoHandlerProvider
+     * @dataProvider facetWithInvalidModeProvider
+     * @dataProvider missingFilterProvider
+     *
+     * @param SimpleXMLElement $xmlResponse
+     * @param array $expectedResults
+     *
+     * @throws Exception
+     */
+    public function testProductNumberSearchResultWithAllFilters(SimpleXMLElement $xmlResponse, array $expectedResults)
+    {
+        $xml = $xmlResponse->asXML();
+
+        $criteria = new Criteria();
+        $criteria->setFetchCount(true);
+
+        Shopware()->Session()->findologicDI = false;
+        Shopware()->Session()->isSearchPage = true;
+
+        $mockedQuery = $this->getMockBuilder(QueryBuilder::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['execute'])
+            ->getMockForAbstractClass();
+
+        $mockedQuery->expects($this->once())->method('execute')->willReturn($xml);
+
+        // Mock querybuilder factory method to check that custom implementation does not get called
+        // as original implementation will be called in this case
+        $mockQuerybuilderFactory = $this->createMock(QueryBuilderFactory::class);
+        $mockQuerybuilderFactory->expects($this->once())
+            ->method('createProductQuery')
+            ->willReturn($mockedQuery);
+
+        $originalService = $this->createMock(ProductNumberSearch::class);
+        $productNumberSearch = new ProductNumberSearch($originalService, $mockQuerybuilderFactory);
+
+        $request = new RequestHttp();
+        $request->setModuleName('frontend');
+
+        // Create Mock object for Shopware Front Request
+        $front = $this->getMockBuilder(Front::class)
+            ->setMethods(['Request'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $front->method('Request')->willReturn($request);
+
+        $hydrator = new CustomListingHydrator();
+        $customFacets = [];
+
+        foreach ($xmlResponse->filters->filter as $filter) {
+            $facet = $hydrator->hydrateFacet($filter);
+            $customFacets[] = $facet->getFacet();
+        }
+
+        foreach ($customFacets as $customFacet) {
+            $criteria->addFacet($customFacet);
+        }
```

Comment:
Why do we have to loop through `$customFacets`? Doing `$criteria->addFacet($facet->getFacet());` in the `foreach` above seems to be sufficient.

Reply:
Ahan yes indeed it is. Thanks.